Source/WebCore/rendering/MotionPath.cpp

```diff
@@ -53,20 +53,11 @@ std::optional<MotionPathData> MotionPath::motionPathDataForRenderer(const Render
         return floatPointForLengthPoint(offsetPosition, referenceRect.size());
     };
 
-    if (is<BoxPathOperation>(pathOperation)) {
+    if (is<ShapePathOperation>(pathOperation) || is<BoxPathOperation>(pathOperation)) {
         if (auto* container = renderer.containingBlock()) {
-            auto& boxPathOperation = downcast<BoxPathOperation>(*pathOperation);
-            data.containingBlockBoundingRect = snapRectToDevicePixelsIfNeeded(container->referenceBoxRect(boxPathOperation.referenceBox()), downcast<RenderLayerModelObject>(renderer));
-            data.offsetFromContainingBlock = offsetFromContainer(renderer, *container, data.containingBlockBoundingRect);
-            return data;
-        }
-        return std::nullopt;
-    }
-    if (is<ShapePathOperation>(pathOperation)) {
-        if (auto* container = renderer.containingBlock()) {
-            auto& shapePathOperation = downcast<ShapePathOperation>(*pathOperation);
-            data.containingBlockBoundingRect = snapRectToDevicePixelsIfNeeded(container->referenceBoxRect(shapePathOperation.referenceBox()), downcast<RenderLayerModelObject>(renderer));
-            data.offsetFromContainingBlock = offsetFromContainer(renderer, *container, data.containingBlockBoundingRect);
+            auto referenceRect = snapRectToDevicePixelsIfNeeded(container->referenceBoxRect(pathOperation->referenceBox()), downcast<RenderLayerModelObject>(renderer));
+            data.containingBlockBoundingRect = FloatRoundedRect(container->style().getRoundedBorderFor(LayoutRect(referenceRect)));
```

Comment:
The spec says:
> If `<offset-path>` is omitted, it defaults to inset(0 round X), where X is the value of [border-radius](https://drafts.csswg.org/css-backgrounds-3/#propdef-border-radius) on the element that establishes the [containing block](https://drafts.csswg.org/css-display-4/#containing-block) for this element. If [<coord-box>](https://drafts.csswg.org/css-box-4/#typedef-coord-box) is omitted, it defaults to border-box.

Can we add WPTs that visually this does not affect ShapePathOperation? I know it doesn't right now, but it would be good still to prevent potential regressions.

Reply:
(I'm also thinking maybe we should be intentional about only passing the border-radius in the `BoxPathOperation` case, since we're intentionally not passing it for `RayPathOperation`)

Reply:
Added. Also refactored the code to only pass for BoxPathOperation.